patches/chromium/chore_modify_chromium_handling_of_mouse_events.patch

```diff
@@ -0,0 +1,161 @@
+     }
+     // We must let Windows handle the caption buttons if it's drawing them, or
+     // they won't work.
++    bool simulate_mouse_event_for_caption = false;
+     if (delegate_->GetFrameMode() == FrameMode::SYSTEM_DRAWN &&
+         (hittest == HTCLOSE || hittest == HTMINBUTTON ||
+          hittest == HTMAXBUTTON)) {
+-      SetMsgHandled(FALSE);
++      simulate_mouse_event_for_caption =
++          delegate_->HandleMouseEventForCaption(message);
++      if (!simulate_mouse_event_for_caption)
++        SetMsgHandled(FALSE);
+     }
+     // Let resize events fall through. Ignore everything else, as we're either
+     // letting Windows handle it above or we've already handled the equivalent
+     // touch message.
+-    if (!IsHitTestOnResizeHandle(hittest)) {
++    if (!IsHitTestOnResizeHandle(hittest) && !simulate_mouse_event_for_caption) {
+       return 0;
+     }
+   }
+@@ -3172,11 +3176,21 @@ LRESULT HWNDMessageHandler::HandleMouseEventInternal(UINT message,
+     message = WM_MOUSEHWHEEL;
+   }
+ 
++  LONG message_time = GetMessageTime();
++  CHROME_MSG msg = {hwnd(),
++                    message,
++                    w_param,
++                    l_param,
++                    static_cast<DWORD>(message_time),
++                    {CR_GET_X_LPARAM(l_param), CR_GET_Y_LPARAM(l_param)}};
++  ui::MouseEvent event(msg);
++  bool sys_menu_handled = false;
+   if (message == WM_RBUTTONUP && is_right_mouse_pressed_on_caption_) {
+     // TODO(pkasting): Maybe handle this in DesktopWindowTreeHostWin, where we
+     // handle alt-space, or in the frame itself.
+     is_right_mouse_pressed_on_caption_ = false;
+     ReleaseCapture();
++
+     // |point| is in window coordinates, but WM_NCHITTEST and TrackPopupMenu()
+     // expect screen coordinates.
+     POINT screen_point = CR_POINT_INITIALIZER_FROM_LPARAM(l_param);
+@@ -3184,7 +3198,10 @@ LRESULT HWNDMessageHandler::HandleMouseEventInternal(UINT message,
+     w_param = static_cast<WPARAM>(SendMessage(
+         hwnd(), WM_NCHITTEST, 0, MAKELPARAM(screen_point.x, screen_point.y)));
+     if (w_param == HTCAPTION || w_param == HTSYSMENU) {
+-      ShowSystemMenuAtScreenPixelLocation(hwnd(), gfx::Point(screen_point));
++      event.set_is_system_menu(true);
++      right_click_handled = delegate_->HandleMouseEvent(&event);
++      if (!right_click_handled)
++        ShowSystemMenuAtScreenPixelLocation(hwnd(), gfx::Point(screen_point));
+       return 0;
+     }
+   } else if (message == WM_NCLBUTTONDOWN &&
+@@ -3219,14 +3236,6 @@ LRESULT HWNDMessageHandler::HandleMouseEventInternal(UINT message,
+     SetCapture();
+   }
+ 
+-  LONG message_time = GetMessageTime();
+-  CHROME_MSG msg = {hwnd(),
+-                    message,
+-                    w_param,
+-                    l_param,
+-                    static_cast<DWORD>(message_time),
+-                    {CR_GET_X_LPARAM(l_param), CR_GET_Y_LPARAM(l_param)}};
+-  ui::MouseEvent event(msg);
```

Comment:
`w_param` in this case is different from the system menu case, should we avoid hoisting this block. Instead leave this as it is and make the following changes